File: LeetCode_Solutions/LeetCode_Solutions/DetectCapital_520.m
//
//  DetectCapital_520.m
//  LeetCode_Solutions
//
//  Created by Huni on 31/03/2017.
//  Copyright © 2017 KnightJoker. All rights reserved.
//

#import "DetectCapital_520.h"

@implementation DetectCapital_520

- (BOOL)detectCapitalUse:(NSString *)word {

    NSInteger count = 0;

    for (NSInteger i = 0; i < word.length; i++) {
        if ([word characterAtIndex:i] > 'a' && [word characterAtIndex:i] < 'z') {
            count ++;
        }
    }
    
    if (count == word.length || count == 0) {
        return YES;
    } else if (count == word.length - 1 && [word characterAtIndex:0] > 'A' && [word characterAtIndex:0] < 'Z'){
        return YES;
    } else {
        return NO;
    }
}

@end

File: LeetCode_Swift/LeetCode_Swift/DetectCapital_520.swift
//
//  DetectCapital_520.swift
//  LeetCode_Swift
//
//  Created by Huni on 28/08/2017.
//  Copyright © 2017 KnighhtJoker. All rights reserved.
//

import Foundation

public class DetectCapital_520 {

    func detectCapitalUse(_ word: String) -> Bool {
        
        var count = 0
        
        for char in word {
            
            if char >= "a" && char <= "z" {
                count += 1
            }
        }
        
        if count == word.count || count == 0 {
            return true
        } else if count == word.count - 1 && word[word.startIndex] >= "A" && word[word.startIndex] <= "Z" {
            return true
        } else {
            return false
        }

    }
}
